### PR TITLE
fix: rebuild lr scheduler when critic warmup finished

### PR DESCRIPTION
**Pull request by anHappyDog**
### Description
This PR fixes an FSDP checkpoint resume issue that can occur after critic warmup. During critic warmup, the optimizer is initially built for the value head only; once warmup finishes, the optimizer is rebuilt for the full trainable parameter set, but the LR scheduler was still bound to the old optimizer. This could produce inconsistent optimizer param group state, including missing `initial_lr` keys during DCP checkpoint load. The fix rebuilds the LR scheduler whenever the optimizer is rebuilt after critic warmup, and adds an explicit `last_epoch` argument to `build_lr_scheduler` with the default `-1`, so the post-warmup scheduler starts fresh for the new optimizer. Verification: `python -m py_compile rlinf/hybrid_engines/fsdp/fsdp_model_manager.py` and `ruff check rlinf/hybrid_engines/fsdp/fsdp_model_manager.py`.

### Motivation and Context
Fix resume.

### How has this been tested?
Test save and resume for pi and openvlaoft.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (Document-only update)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by andylin-hao — APPROVED** [review]

File: rlinf/hybrid_engines/fsdp/fsdp_model_manager.py [hunk 1]
```diff
@@ -441,13 +441,17 @@ def optimizer_step(self) -> tuple[float, list[float]]:
             if self.optimizer_steps >= self.critic_warmup_steps:
                 self.optimizer = self.build_optimizer(model=self.model)
                 self.critic_warmup_steps = 0
+                self.lr_scheduler = self.build_lr_scheduler(
+                    optimizer=self.optimizer,
+                    optim_config=self._cfg.optim,
```
Comment (qurakchin) [inline rlinf/hybrid_engines/fsdp/fsdp_model_manager.py:446]: should the last_epoch be added here?

Comment (anHappyDog) [reply rlinf/hybrid_engines/fsdp/fsdp_model_manager.py:446]: no need, here I think the warmup LR schedule does not need to be continued. If we implement it so that the LR schedule during warmup should continue, it becomes more complicated, requiring manual setting of the initial LR and handling of additional scheduler steps.

**Review by guozhen1997 — APPROVED** [review]
